ext/spl/spl_array.c

```diff
@@ -304,6 +305,14 @@ static zval *spl_array_get_dimension_ptr(int check_inherited, spl_array_object *
 		return &EG(error_zval);
 	}
 
+	if (check_inherited && intern->fptr_offset_get) {
+        zend_call_method_with_1_params(object, Z_OBJCE_P(object), &intern->fptr_offset_get, "offsetGet", retval, offset);
```

Comment:
You're passing NULL as the retval, so the retval will be ignored. Please modify your test so that this is visible, e.g. with extra var dumps after the indirect modifications.

You need to pass a pointer to a zval here. Note that this is not actually possible while using `get_property_ptr_ptr`, which can only return owned storage. It's only possible to perform calls like this from `read_property`. So what you want to do is trigger a read_property fallback in get_property_ptr_ptr if offsetGet has been overridden.